swc.config.ts

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable @typescript-eslint/no-require-imports */
+const { swcDir } = require("@swc/cli");
+import type { Options } from "@swc/core";
+import path from "path";
+
+/**
+ * This script configures SWC programatically to compile the project.
+ *
+ * The main reason for doing this over using .swcrc is to be able to
+ * dynamically set feature flags based on environment variables. These
+ * feature flags can then be used in the code to conditionally compile
+ * different code paths.
+ *
+ * These are not neccessarily user facing features, they are more like
+ * infrastructure within the package. An example could be to enable
+ * debug logging in development, or to enable fetch with credentials
+ * during development.
+ *
+ * A lof this was generated in conversation with Uncle Claude, in case
+ * you were wondering...
+ *
+ */
+
+// Define all possible feature flags. They key will be turned into a
+// global variable wrapped with double underscores, e.g. __DEBUG_MODE__,
+// and the value will be the environment variable to read.
+const FEATURE_FLAGS = {
+  ENABLE_FETCH_WITH_CREDENTIALS: "ENABLE_FETCH_WITH_CREDENTIALS",
+  // DEBUG_MODE: 'DEBUG_MODE',
+} as const;
+
+// Helper to convert environment variable to boolean
+const envToBool = (value: string | undefined): boolean => {
+  if (!value) return false;
+  return ["1", "true", "yes"].includes(value.toLowerCase());
+};
+
+// Get all feature flags from environment
+const getFeatureFlags = () => {
+  const flags: Record<string, string> = {};
+
+  Object.entries(FEATURE_FLAGS).forEach(([key, envVar]) => {
+    // Convert environment variable values to string 'true' or 'false'
+    // This matches the format SWC expects in its globals configuration
+    flags[`__${key}__`] = String(envToBool(process.env[envVar]));
```

Comment:
I was going to suggest changing the context of `envToBool` to return a string directly since this is the only place this is used, but upon reflection this doesn't actually make things any simpler.

Reply:
Same with re-expressing this `forEach` as a `.reduce` on `Object.entries(FEATURE_FLAGS)` to build the object as you go and returning directly. A stylistic suggestion but completely optional.